src/main/java/com/epam/ta/reportportal/demo_data/DemoDashboardsService.java

```diff
@@ -0,0 +1,136 @@
+import com.epam.ta.reportportal.database.search.Filter;
+import com.epam.ta.reportportal.exception.ReportPortalException;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+
+@Service
+class DemoDashboardsService {
+	private static final String DEMO_WIDGETS = "demo_widgets.json";
+	private UserFilterRepository userFilterRepository;
+	private DashboardRepository dashboardRepository;
+	private WidgetRepository widgetRepository;
+
+	@Autowired
+	DemoDashboardsService(UserFilterRepository userFilterRepository, DashboardRepository dashboardRepository,
+			WidgetRepository widgetRepository) {
+		this.userFilterRepository = userFilterRepository;
+		this.dashboardRepository = dashboardRepository;
+		this.widgetRepository = widgetRepository;
+	}
+
+	Dashboard createDemoDashboard(DemoDataRq rq, String user, String project) {
+		final Dashboard dashboard = dashboardRepository.findOneByUserProject(user, project, rq.getDashboardName());
+		if (dashboard != null) {
+			throw new ReportPortalException(
+					"Dashboard with name " + rq.getDashboardName() + " already exists. You couldn't create the duplicate.");
+		}
+		String filterId = createDemoFilter(rq.getFilterName(), user, project);
+		Acl acl = acl(user, project);
+		try {
+			final URL resource = this.getClass().getClassLoader().getResource(DEMO_WIDGETS);
+			if (resource == null) {
+				throw new ReportPortalException("Unable to find demo_widgets.json");
+			}
+			try (InputStreamReader json = new InputStreamReader(new FileInputStream(resource.getPath()), UTF_8)) {
+				Type type = new TypeToken<List<Widget>>() {
+				}.getType();
+				List<Widget> widgets = ((List<Widget>) new Gson().fromJson(json, type)).stream().map(it -> {
+					it.setProjectName(project);
+					it.setApplyingFilterId(filterId);
+					it.setAcl(acl);
+					return it;
+				}).collect(toList());
+				List<Widget> save = widgetRepository.save(widgets);
+				for (Widget widget : save) {
+					System.out.println(widget.getId());
+				}
+				return createDemoDashboard(widgets, user, project, rq.getDashboardName());
+			}
+		} catch (IOException e) {
+			throw new ReportPortalException("Unable to load demo_widgets.json");
+		}
+	}
+
+	private String createDemoFilter(String filterName, String user, String project) {
+		UserFilter userFilter = new UserFilter();
+		userFilter.setName(filterName);
+		userFilter.setFilter(new Filter(Launch.class, HAS, false, "demo", "tags"));
+		SelectionOptions selectionOptions = new SelectionOptions();
+		selectionOptions.setSortingColumnName("start_time");
+		selectionOptions.setIsAsc(false);
+		selectionOptions.setQuantity(50);
+		selectionOptions.setPageNumber(1);
+		userFilter.setSelectionOptions(selectionOptions);
+		userFilter.setProjectName(project);
+		userFilter.setIsLink(false);
+		Acl acl = acl(user, project);
+		userFilter.setAcl(acl);
+		UserFilter existingFilter = userFilterRepository.findOneByName(user, filterName, project);
```

Comment:
I'd like to suggest making this check first, and then create the instance of UserFilter pojo, because currently it seems a bit confusing (create new entity -> check if already created -> save)